server/src/infra/ldap_server.rs

```diff
@@ -39,12 +39,22 @@ where
             if result.is_empty() {
                 debug!("No response");
             }
+            let results: i64 = result.len().try_into().unwrap();
             for response in result.into_iter() {
                 debug!(?response);
+                let controls =
```

Comment:
If I wanted to be pedantic, I would ask you to assert that it's the last element in the response.

Thankfully, I'll have to maintain that code, and the simpler the better ;) I like this solution.